spec/support/synchronization_helpers.rb

```diff
@@ -40,17 +40,6 @@ def expect_in_fork(fork_expectations: nil)
     end
   end
 
-  def expect_in_thread(&block)
-    # Start in thread
-    t = Thread.new(&block)
-
-    # Wait for thread to finish, retrieve its return value.
-    status = t.value
-
-    # Expect thread and assertions to have completed successfully.
-    expect(status).to be true
-  end
-
```

Comment:
I added this while revamping the `synchronization_helpers`, but I was mistaken thinking that one would find it useful to assert on a Thread's return value like this, especially a simple `status == true` assertion.

The idea here was borrowed from `expect_in_fork`, but that one has a much clearer exit code to check on, and the separate memory space of forks makes that variant much more useful.